src/wallet/asyncrpcoperation_common.cpp

```diff
@@ -40,3 +41,108 @@ std::pair<CTransaction, UniValue> SignSendRawTransaction(UniValue obj, std::opti
+                default:
+                    assert(false);
+            }
+        },
+        [&](const InvalidFundsError& err) {
+            bool isFromUa = std::holds_alternative<libzcash::UnifiedAddress>(selector.GetPattern());
+            throw JSONRPCError(
+                RPC_INVALID_PARAMETER,
+                strprintf(
+                    "Insufficient funds: have %s, %s",
+                    FormatMoney(err.available),
+                    std::visit(match {
+                        [](const InsufficientFundsError& ife) {
+                            return strprintf("need %s", FormatMoney(ife.required));
+                        },
+                        [](const DustThresholdError& dte) {
+                            return strprintf(
+                                    "need %s more to avoid creating invalid change output %s (dust threshold is %s)",
+                                    FormatMoney(dte.dustThreshold - dte.changeAmount),
+                                    FormatMoney(dte.changeAmount),
+                                    FormatMoney(dte.dustThreshold));
+                        }
+                    },
+                    err.reason))
+                    + (selector.TransparentCoinbasePolicy() != TransparentCoinbasePolicy::Disallow
+                       ? "" :
+                       "; note that coinbase outputs will not be selected if you specify "
+                       "ANY_TADDR, any transparent recipients are included, or if the "
+                       "`privacyPolicy` parameter is not set to `AllowRevealedSenders` or weaker")
+                    + (!isFromUa || strategy.AllowLinkingAccountAddresses() ? "." :
+                       ". (This transaction may require selecting transparent coins that were sent "
+                       "to multiple Unified Addresses, which is not enabled by default because "
+                       "it would create a public link between the transparent receivers of these "
+                       "addresses. THIS MAY AFFECT YOUR PRIVACY. Resubmit with the `privacyPolicy` "
+                       "parameter set to `AllowLinkingAccountAddresses` or weaker if you wish to "
+                       "allow this transaction to proceed anyway.)"));
+        },
+        [](const ChangeNotAllowedError& err) {
+            throw JSONRPCError(
+                    RPC_WALLET_ERROR,
+                    strprintf(
+                        "When shielding coinbase funds, the wallet does not allow any change. "
+                        "The proposed transaction would result in %s in change.",
+                        FormatMoney(err.available - err.required)));
+        },
+        [](const ExcessOrchardActionsError& err) {
+            throw JSONRPCError(
+                RPC_INVALID_PARAMETER,
+                strprintf(
+                    "Attempting to spend %u Orchard notes would exceed the current limit "
```

Comment:
This error is also returned when too many Orchard recipients would be needed; the error message should be adjusted in that case.